src/window_service.rs

```diff
@@ -144,7 +145,7 @@ fn recv_window(
     consumed: &mut u64,
     received: &mut u64,
     r: &BlobReceiver,
-    s: &BlobSender,
+    s: &Sender<Vec<Entry>>,
```

Comment:
Following the "When in Rome" philosophy mentioned in CONTRIBUTING.md, you need to either add an `EntrySender` type alias somewhere or get rid of the `BlobReceiver` alias. No mixing of coding styles within a single module please.